scaleway/data_source_object_bucket_test.go

```diff
@@ -38,3 +44,191 @@ func TestAccScalewayDataSourceObjectStorage_Basic(t *testing.T) {
 		},
 	})
 }
+
+func TestAccScalewayDataSourceObjectStorage_ProjectIDAllowed(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	bucketName := sdkacctest.RandomWithPrefix("test-acc-scaleway-object-bucket")
+
+	project, iamAPIKey, terminateFakeSideProject, err := createFakeSideProject(tt)
+	require.NoError(t, err)
+	defer terminateFakeSideProject()
+
+	ctx := context.Background()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: fakeSideProjectProviders(ctx, tt, project, iamAPIKey),
+		CheckDestroy:      testAccCheckScalewayObjectDestroy(tt),
+		Steps: []resource.TestStep{
+			// Create a bucket from the main provider into the side project and read it from the side provider
+			// The side provider should only be able to read the bucket from the side project
+			{
+				Config: fmt.Sprintf(`
```

Comment:
Please add this example on the resource doc with explanation

Reply:
This is tricks using a provider to do the test, we are implementing `project_id` to avoid doing this